app/scripts/metamask-controller.js

```diff
@@ -1210,6 +1210,8 @@ export default class MetamaskController extends EventEmitter {
           ?.disabledRpcMethodPreferences?.eth_sign,
       getAllState: this.getState.bind(this),
       securityProviderRequest: this.securityProviderRequest.bind(this),
+      getCurrentChainId: () =>
+        this.networkController.store.getState().providerConfig.chainId,
```

Comment:
I assume this is guaranteed to always exist so no risk of an undefined `providerConfig`?

Reply:
Almost all controllers determine `chainId` this way, so I assume this is kinda guaranteed.